var/spack/repos/builtin/packages/wrf/package.py

```diff
@@ -15,132 +60,309 @@ class Wrf(Package):
-                      "pgi":   {"serial": '52',
-                                "smpar":  '53',
-                                "dmpar":  '54',
-                                "dmsm":   '55'},
-                      }
-
-        nesting_opts = {"basic":  "1",
-                        "preset": "2",
-                        "vortex": "3"}
-
-        try:
-            compiler_opts = build_opts[self.spec.compiler.name]
-        except KeyError:
-            raise InstallError("Compiler not recognized nor supported.")
-
-        # Spack already makes sure that the variant value is part of the set.
-        build_type = compiler_opts[spec.variants['build_type'].value]
-
-        nesting_value = nesting_opts[spec.variants['nesting'].value]
-
-        with tempfile.TemporaryFile(mode='w') as fp:
-            fp.write(build_type + '\n' + nesting_value + '\n')
-            fp.seek(0)
-            Executable('./configure')(input=fp)
+
+        # Remove broken default options...
+        self.do_configure_fixup()
+
+        if self.spec.compiler.name not in ["intel", "gcc"]:
+            raise InstallError(
+                "Compiler %s not currently supported for WRF build."
+                % self.spec.compiler.name
+            )
+
+        p = Popen("./configure", stdin=PIPE, stdout=PIPE, stderr=PIPE)
+        setNonBlocking(p.stdout)
+        setNonBlocking(p.stderr)
+
+        # Because of WRFs custom configure scripts that require interactive
+        # input we need to parse and respond to questions.  The details can
+        # vary somewhat with the exact version, so try to detect and fail
+        # gracefully on unexpected questions.
+        stallcounter = 0
+        outputbuf = ""
+        while True:
+            line = p.stderr.readline().decode()
+            if not line:
+                line = p.stdout.readline().decode()
+            if not line:
+                if p.poll() is not None:
+                    returncode = p.returncode
+                    break
+                if stallcounter > 300:
+                    raise InstallError(
+                        "Output stalled for 30s, presumably an "
+                        "undetected question."
+                    )
+                time.sleep(0.1)  # Try to do a bit of rate limiting
+                stallcounter += 1
+                continue
+            stdout.write(line)
+            stallcounter = 0
+            outputbuf += line
+            if (
+                "Enter selection" in outputbuf
+                or "Compile for nesting" in outputbuf
+            ):
+                answer = self.answer_configure_question(outputbuf)
+                p.stdin.write(answer.encode())
+                p.stdin.flush()
+                outputbuf = ""
+
+        if returncode != 0:
+            raise InstallError("Configure failed - unknown error")
+
+    def run_compile_script(self):
+        csh_bin = path_join(self.spec["tcsh"].prefix, "bin/csh")
+
+        compilation = Popen(
```

Comment:
Instead of using `Popen`, is it possible to use Spack's `Executable` class? Something like:
```python
csh = Executable(csh_bin)
csh(...)
```

Reply:
Having looked at the source for the Executable class I think I can, but the output from that compile stage is massive, so I was trying to avoid keeping the whole build log in RAM.  If spack is doing that anyway behind the scenes then I think I can convert this one to `Executable` but not the interactive configure step.

Reply:
Ah, I don't think we're doing that, didn't realize this package needed to be so complicated.

Reply:
Testing with Executable gives me the behaviour I need, thanks!